src/main/webapp/authorization.js

```diff
@@ -1,56 +1,124 @@
+     * GitHubAuthorizer
+     * @type {Firebase} 
+     */
+    this.firebase = firebase; // eslint-disable-line
+
+    this.initializeFirebase()
   }
-}
 
-/**
- * Get firebase
- * @return {Firebase}
- */
-function getFirebase() {
-  return firebase; //eslint-disable-line
-}
+  /**
+   * The most recent access token for the GitHub API. To refresh this, the user
+   * must be signed out and signed in again. It is important to keep in mind
+   * that this token expires and must be used swiftly after calling
+   * toggleSignIn(). If authorization is yet to be toggled, this will return
+   * null. If an error occurs during authorization, it will again be set to
+   * null. This property will be set to null on sign out as well.
+   *
+   * NOTE: In the event that an access token times out, the token will still be
+   * available here and so, a non-null token is not a good sign of a valid
+   * token.
+   *
+   * @return {string} most recent token or null
+   */
+  getToken() {
+    return this.token;
+  }
 
+  /**
+   * The current user if authorized and null if not. For all intents and
+   * purposes, this function will serve as a true/false value in an if
+   * statement.
+   *
+   * @return {Firebase.User} current user or null
+   */
+  getUser() {
+    return this.firebase.auth().currentUser;
+  }
 
-/**
- * Function called when clicking the Login/Logout button.
- */
-function toggleSignIn() {
-  const firebase = getFirebase();
-  if (!firebase.auth().currentUser) {
-    let provider = new firebase.auth.GithubAuthProvider();
-    firebase.auth().signInWithPopup(provider).then(function(result) {
-      // This gives you a GitHub Access Token.
-      // You can use it to access the GitHub API.
-      const token = result.credential.accessToken; //eslint-disable-line
-      // The signed-in user info.
-      const user = result.user; //eslint-disable-line
-    }).catch(function(error) {
-      // Handle Errors here.
-      console.error(error);
-      // Change this in future iterations
-      alert(error.message);
-    });
-  } else {
-    // if logged in, sign out
-    firebase.auth().signOut();
+  /**
+   * Get firebase instance used with this object
+   * @return {Firebase} firebase instance
+   */
+  getFirebase() {
+    return this.firebase;
+  }
+
+  /**
+   * Function called when authorizing user to verify them as a gitHub user.
+   * This function doubles as an opt out when the current user wants to leave
+   * our site for good.
+   *
+   * Side Effects: when authorizing a user, it sets the current objects token
+   * to the returned GitHub API accessToken if a token is returned.
+   *
+   * NOTE: this function is asynchronous and should be used with an await
+   *
+   * @return {boolean} true if no errors occured during authorization, false
+   * otherwise
+   */
+  async toggleAuthorization() {
+    if (!this.getFirebase().auth().currentUser) {
+      try {
+        let provider = new this.firebase.auth.GithubAuthProvider();
```

Comment:
I'm getting an error from this line that corresponds to [this SO post](https://stackoverflow.com/questions/39397381/react-firebase-googleauthprovider-is-not-a-constructor). However, I'm having a lot of trouble fixing it even though the solution is quite clear. Let me know if you can figure it out.